library/src/main/java/com/mux/player/internal/cache/CacheDatastore.kt

```diff
@@ -198,15 +246,73 @@ internal class CacheDatastore(val context: Context) : Closeable {
     return key
   }
 
-  /**
-   * Generates a URL-safe cache key for a given URL. Delegates to [generateCacheKey] but encodes it
-   * into something else
-   */
-  @JvmSynthetic
-  internal fun safeCacheKey(url: URL): String = Base64.encodeToString(
-    generateCacheKey(url).toByteArray(Charsets.UTF_8),
-    Base64.NO_WRAP or Base64.URL_SAFE
-  )
+  private fun doDeleteRecords(db: SQLiteDatabase, records: List<FileRecord>): Result<Int> {
+    return runCatching {
+      var written = 0
+      records.map { it.lookupKey }.forEach {
+        written += db.delete(
+          /* table = */ IndexSql.Files.name,
+          /* whereClause = */ "${IndexSql.Files.Columns.lookupKey}=?",
+          /* whereArgs = */ arrayOf(it)
+        )
+      }
+      written
+    }
+  }
+
+  private fun doEvictByLru(db: SQLiteDatabase): Result<Int> {
+    try {
```

Comment:
Are db operations run in their own (effectively non-blocking) context?

Reply:
db operations block, but it's good that this happens, and that question doesn't map cleanly from iOS onto Android 

Blocking for I/O happens out in the open in java/kotlin (with these apis, etc etc), and you decide when/how to block, and also when/how to dispatch async work. Asynchronous contexts are built this by app and library developers. Here, we're already on a player thread that is expected to block for I/O and return when done, and so we can just do our work synchronously.

In our case, the only times that two player threads would block for access to the cache is during the initial lookup or when finalizing the write. When a file is being streamed to/from the disk, nothing interferes. The overall contribution to latency will be like a handful of millis and only under certain scenarios.

Hope that was helpful!